pyked/converters.py

```diff
@@ -0,0 +1,522 @@
+        file_author_orcid (`str`, optional): ORCID of file author
+    """
+    from pyked.chemked import ChemKED
+
+    assert os.path.isfile(filename_xml), 'Error: ' + filename_xml + ' file missing'
+
+    # get all information from XML file
+    tree = etree.parse(filename_xml)
+    root = tree.getroot()
+
+    # get file metadata
+    properties = get_file_metadata(root)
+
+    # get reference info
+    properties['reference'] = get_reference(root)
+    # Save name of original data filename
+    properties['reference']['detail'] = (properties['reference'].get('detail', '') +
+                                         'Converted from ReSpecTh XML file ' +
+                                         os.path.basename(filename_xml)
+                                         )
+
+    # Ensure ignition delay, and get which kind of experiment
+    properties.update(get_experiment_kind(root))
+
+    # Get properties shared across the file
+    properties['common-properties'] = get_common_properties(root)
+
+    # Determine definition of ignition delay
+    properties['common-properties']['ignition-type'] = get_ignition_type(root)
+
+    # Now parse ignition delay datapoints
+    properties['datapoints'] = get_datapoints(root)
+
+    # Ensure inclusion of pressure rise or volume history matches apparatus.
+    if ('pressure-rise' in properties['common-properties'] or
+        any([dp for dp in properties['datapoints'] if dp.get('pressure-rise')])
+        ) and properties['apparatus']['kind'] == 'rapid compression machine':
+        raise KeywordError('Pressure rise cannot be defined for RCM.')
+
+    if ('volume-history' in properties['common-properties'] or
+        any([dp for dp in properties['datapoints'] if dp.get('volume-history')])
+        ) and properties['apparatus']['kind'] == 'shock tube':
+        raise KeywordError('Volume history cannot be defined for shock tube.')
+
+    # apply any overrides
+    if file_author:
+        properties['file-author']['name'] = file_author
+    if file_author_orcid:
+        properties['file-author']['ORCID'] = file_author_orcid
+
+    # Now go through datapoints and apply common properties
+    for idx in range(len(properties['datapoints'])):
+        for prop in properties['common-properties']:
+            properties['datapoints'][idx][prop] = properties['common-properties'][prop]
+
+    # compression time doesn't belong in common-properties
+    properties['common-properties'].pop('compression-time', None)
+
+    # set output filename and path
+    if not filename_ck:
+        filename_ck = os.path.splitext(os.path.basename(filename_xml))[0] + '.yaml'
+
+    with open(filename_ck, 'w') as outfile:
+        outfile.write(yaml.dump(properties, default_flow_style=False))
+    print('Converted to ' + filename_ck)
+
+    # now validate
+    ChemKED(yaml_file=filename_ck)
+
+
+def main(argv):
+    """
+    """
+    from pyked.chemked import ChemKED
```

Comment:
Same issue as before—a module-level import would lead to circular import, I think